nvdaHelper/espeak/sconscript

```diff
@@ -84,25 +137,198 @@ def espeak_compilePhonemeData_buildAction(target,source,env):
 
 env['BUILDERS']['espeak_compilePhonemeData']=Builder(action=env.Action(espeak_compilePhonemeData_buildAction,"Compiling phoneme data"),emitter=espeak_compilePhonemeData_buildEmitter)
 
-def espeak_compileDict_buildAction(target,source,env):
+#: See dictionaries section of /include/espeak/Makefile.am
+espeakDictionaryCompileList: typing.Dict[
+	str, # expected dict file name EG 'es_dict'
+	typing.Tuple[str, typing.List[str]] # language code, list of input files
+] = {
+	"af_dict": ("af", ["af_list", "af_rules", ]),

```

Comment:
would it be possible to generate `espeakDictionaryCompileList` by 

- mapping the lang codes (eg `("af", "am", ...)` to each default entry eg `"af_dict": ("af", ["af_list", "af_rules", ]`, and then 
- afterwards, make the changes to the unique entries by appending to the list of input files 

I think it would make the unique entries a bit easier to track.

Reply:
I'm not sure what you mean by unique entries?

Previously this list was essentially generated, we used glob to find all the *_rules files, split the file name using the '_' character to get the language code. We then said that any file starting with that language code is a dependency of the <lang Code>_dict file.

Explicitly listing the files means that the file names don't have to correspond to the language. This WAS the case for zhy_rules, which has now been fixed in espeak-ng to become yue_rules.

Note that espeak's makefile lists this explicitly as well, so now with an update we just reflect the changes they make to the makefile. It forces us to be more aware of changes in language support.

Reply:
I was suggesting something of the following form, that is still explicit, but perhaps easier to parse and understand why/how it is generated. 

```python
espeakLangCodes = ("af", "am", ...)
espeakDictionaryCompileList = {}
for code in espeakLangCodes:
     espeakDictionaryCompileList[f"{code}_dict"] = ("{code}", ["{code}_list", "{code}_rules", ])
# perform further changes here
espeakDictionaryCompileList["cmn_dict"][1].append("cmn_listx")
espeakDictionaryCompileList["yue_dict"][1][1].replace("yue_rules", "zhy_rules")  # unnecessary now
...
```

Reply:
We could do it like that. I was trying to optimize it for replicating changes to the espeak `makefile` which has entries like this:
```
el: espeak-ng-data/el_dict
espeak-ng-data/el_dict: dictsource/el_list dictsource/el_rules dictsource/el_extra dictsource/el_emoji
```

We specifically delete the `emoji` files, we prefer to get this data from the `cldr` library. I'm not sure about `extra`.

Reply:
As someone unfamiliar with this process, I would go with whatever is easier for the process. I assume you determine the changes by updating `espeak` and introducing the changes from diffing the `makefile`.

Reply:
Yep, exactly. Then reproducing the changes from the diff of the makefile in our sconscript.